### PR TITLE
Make the spider test crawl the sitemap (as good bots do)

### PR DESCRIPTION
**Pull request by coolo**
We no longer have a way to list all projects nor a way to list
all packages in projects without having javascript due to use
of ajax. This made the spider test a game of luck and certain
packages were only found if they were listed in requests or
rebuild graphs (as it happened in Eduardo's PR for some reason).

But to have reproducible result, it's important to crawl all
pages to find also those we don't normally test in rspec

### CODE REVIEW

**Comment by hennevogel** [comment]
Not that any real spider would use our sitemap, but good idea 😊

File: src/api/test/functional/webui/spider_test.rb [hunk 1]
```diff
@@ -7,6 +7,28 @@
 ENV['BOOTSTRAP'] = '1'
 
 class Webui::SpiderTest < Webui::IntegrationTest
+  def ignore_link?(link)
+    return 1 if link =~ %r{/mini-profiler-resources}
```
Comment (hennevogel) [inline src/api/test/functional/webui/spider_test.rb]: nitpick: `return true...`

File: src/api/test/functional/webui/spider_test.rb [hunk 2]
```diff
@@ -7,6 +7,28 @@
...
+    # that link is just a top ref
+    return 1 if link.end_with?('/package/rdiff')
+    # admin can see even the hidden
+    return 1 if link.end_with?('/package/show/HiddenRemoteInstance')
+    return 1 if link =~ %r{/package/show/SourceprotectedProject}
+    # this is crashing (bug)
+    return 1 if link =~ %r{/package/show/UseRemoteInstance}
```
Comment (coolo) [inline src/api/test/functional/webui/spider_test.rb]: Note that this bug is real, but I didn't report it. You can see it in action if you follow any package name on https://build-test.opensuse.org/project/monitor/OBS:Server:Unstable?arch_x86_64=1&broken=1&defaults=0&repo_SLE_12_SP4=1

File: src/api/test/functional/webui/spider_test.rb (hunk 1 above)
Comment (coolo) [reply src/api/test/functional/webui/spider_test.rb]: Yeah, that happens if you mix perl and ruby between day and night :)

Comment (coolo) [reply src/api/test/functional/webui/spider_test.rb]: fixed

**Pull request merged**